### PR TITLE
Implement GetIndexStatistics interface

### PR DESCRIPTION
**Pull request by wayblink**
#22888

### CODE REVIEW

File: internal/proxy/task_index_test.go [hunk 1]
```diff
@@ -600,3 +600,64 @@ func Test_parseIndexParams_AutoIndex(t *testing.T) {
...
+	indexName := "_default"
+	Params.InitOnce()
+	ic := newMockIndexCoord()
+	ctx := context.Background()
+
+	git := getIndexStatisticsTask{
+		ctx: ctx,
+		GetIndexStatisticsRequest: &milvuspb.GetIndexStatisticsRequest{
+			Base: &commonpb.MsgBase{
+				MsgType: commonpb.MsgType_CreateIndex,
```
Comment (czs007) [inline internal/proxy/task_index_test.go:617]: wrong type

File: internal/distributed/indexcoord/client/client.go [hunk 2]
```diff
@@ -203,7 +203,7 @@ func (c *Client) GetIndexInfos(ctx context.Context, req *indexpb.GetIndexInfoReq
 	return ret.(*indexpb.GetIndexInfoResponse), err
 }
 
-// DescribeIndex describe the index info of the collection.
+// DescribeIndex get the statistics of the index.
```
Comment (czs007) [inline internal/distributed/indexcoord/client/client.go:206]: wrong comment

File: internal/distributed/indexcoord/client/client.go [hunk 3]
```diff
@@ -217,6 +217,20 @@ func (c *Client) DescribeIndex(ctx context.Context, req *indexpb.DescribeIndexRe
 	return ret.(*indexpb.DescribeIndexResponse), err
 }
 
+// DescribeIndex describe the index info of the collection.
```
Comment (czs007) [inline internal/distributed/indexcoord/client/client.go:220]: wrong comment

**Comment by xiaofan-luan** [comment]
Let's put it into 2.2.10

**Pull request merged**